libphal/exception.H

```diff
@@ -19,8 +23,68 @@ struct LibphalError : public std::exception {
 	virtual ERR_TYPE errType() const noexcept = 0;
 };
 
+// SBE related functions errors base exception class
+// Exception class includes all the required for SBE RAS
+// specific requirements and PEL.
+// Library users can watch this exception during SBE specific wrapper
+// calls and use the the object specific helper function get the required
+// data for error handling purpose.
+//
+// Note: Application user should remove the file after usage.
+class SbeError final : public LibphalError
+{
+       public:
+	SbeError(ERR_TYPE type, int fd, const char *fileName) :
+	    type(type), fd(fd), fileName(fileName)
+	{
+	}
+	SbeError(ERR_TYPE type) : type(type), fd(-1), fileName(nullptr){};
+
+	virtual ~SbeError(){};
+
+	/* return error type */
+	ERR_TYPE errType() const noexcept override
+	{
+		return type;
+	}
+
+	/* return error string */
+	const char *what() const noexcept override
+	{
+		auto msg = "UNSUPPORTED_ERROR_TYPE";
+		auto errMsg = errMsgMap.find(type);
+		if (errMsg != errMsgMap.end()) {
+			msg = errMsg->second;
+		}
+		return msg;
+	}
+
+	/* return file descriptor includes the FFDC data */
+	int getFd() const noexcept
+	{
+		return fd;
+	}
+
+	/* return filename information */
+	const char *getFileName() const noexcept
+	{
+		return fileName;
+	}
+
+       private:
+	/* Error type */
+	const ERR_TYPE type;
+
+	/* FFDC File descriptor */
+	int fd;
+
+	/* FFDC File name */
+	const char *fileName;
+};
+
 } // namespace exception
 
 using libphalerror_t = exception::LibphalError;
+using sbeerror_t = exception::SbeError;
```

Comment:
Use names which are easily readable - `sbe_error_t` is better than `sbeerror_t`.